pkgs/by-name/dp/dprint/plugins/update-plugins.py

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i python -p nix 'python3.withPackages (pp: with pp; [ requests ])'
+
+import json
+import subprocess
+import requests
+
+
+def nix_prefetch_url(url, algo="sha256"):
+    hash = (
+        subprocess.check_output(["nix-prefetch-url", "--type", algo, url])
+        .decode("utf-8")
+        .rstrip()
+    )
+    sri = (
+        subprocess.check_output(
+            ["nix", "hash", "convert", "--hash-algo", algo, "--to", "sri", hash]
+        )
+        .decode("utf-8")
+        .rstrip()
+    )
+    return sri
+
+
+def update_plugins():
+    data = requests.get("https://plugins.dprint.dev/info.json").json()["latest"]
+    with open("plugins.json", "w+", encoding="utf8") as f:
```

Comment:
You could write the default.nix file directly to avoid doing computation during eval.

Reply:
I read somewhere that generating json is prefered over generating nix, but you are right. I will create a plugins.nix instead.

Reply:
Done, generating the default.nix via update script now.